### PR TITLE
feat: create site from pin on home screen

### LINKED ISSUE #31: Create site from pin on home screen
On the home screen, long-touching to create a pin should prompt to create a site there.

Depends on techmatters/terraso-mobile-client#20 and techmatters/terraso-client-shared#17

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Pins on the home screen now (roughly) match the design & you can create a new site from them. This is ready for review, it is just in draft as a reminder to change the base branch to main before merging, since this PR is stacked on the changes in #188 

### Checklist
- [x] Corresponding issue has been opened
- [x] Change base branch of PR back to main once the PR this is stacked on is merged.

### Related Issues
Closes #31

### CODE REVIEW

File: dev-client/src/components/home/SiteMap.tsx [hunk 1]
```diff
@@ -30,30 +35,92 @@ const siteFeatureCollection = (
...
+            onPress={closeCallout}
+            iconProps={{size: 'md'}}
+          />
+        </Flex>
+        {project && <Heading size="md">{project.name}</Heading>}
+        <Box height="4" />
+        <Box>
+          <Text>
+            {t('site.last_updated', {
+              date: 'dd-mm-yyyy',
```
Comment (paulschreiber) [inline dev-client/src/components/home/SiteMap.tsx:69]: question: would yyyy-mm-dd make sense here?

File: dev-client/src/components/home/SiteMap.tsx [hunk 2]
```diff
@@ -30,30 +35,92 @@ const siteFeatureCollection = (
...
-        <Text>longitude: {site.longitude}</Text>
+      <Box backgroundColor="background.default" padding="4">
+        <Flex direction="row" align="top" justify="space-between">
+          <CalloutDetail label={t('site.soil_id_prediction')} value="???" />
+          <MaterialIconButton name="close" onPress={closeCallout} />
+        </Flex>
+        <Divider />
+        <CalloutDetail
+          label={t('site.ecological_site_prediction')}
+          value="???"
```
Comment (paulschreiber) [inline dev-client/src/components/home/SiteMap.tsx:113]: question: what is the value supposed to be?

File: dev-client/src/translations/en.json [hunk 3]
```diff
@@ -9,8 +9,13 @@
     "members": "{{members}} Members",
     "show_on_map": "SHOW ON MAP",
     "create": "CREATE SITE",
+    "more_info": "MORE INFO",
     "none_existing_p1": "You don’t currently have any saved sites. Create a site by using the button above or by finding the site location on the map.",
-    "none_existing_p2": "A site allows you to collect data, learn more about your land, and share with others. Learn more about using LandPKS..."
+    "none_existing_p2": "A site allows you to collect data, learn more about your land, and share with others. Learn more about using LandPKS...",
```
Comment (paulschreiber) [inline dev-client/src/translations/en.json:14]:
```suggestion
    "none_existing_p2": "A site allows you to collect data, learn more about your land, and share with others. Learn more about using LandPKS.",
```

File: dev-client/src/components/home/SiteMap.tsx (hunk 1 above)
Comment (shrouxm) [reply dev-client/src/components/home/SiteMap.tsx:69]: i am just following the design here, but all these values are placeholders until the actual functionality gets filled out

File: dev-client/src/components/home/SiteMap.tsx (hunk 2 above)
Comment (shrouxm) [reply dev-client/src/components/home/SiteMap.tsx:113]: some output of the soil id algorithm which doesn't exist yet

File: dev-client/src/translations/en.json (hunk 3 above)
Comment (shrouxm) [reply dev-client/src/translations/en.json:14]: i am following the design for this wording, so i think feedback makes more sense to direct there

**Review by david-code — APPROVED** [review]
Looks good! It seems like the temporary site should be transferred into a normal site after creation. Right now it remains a temporary site, and it disappears if you exit the callout. if you  I'm going to create another issue for that.

**Comment by shrouxm** [comment]
@david-code could you clarify? the new site should show up once created because redux refreshes the component's site list. the temporary site i assumed would disappear because you were changing screens, is it not disappearing & that's what you're referring to?

**Pull request merged**